src/OpenTelemetry.Extensions.Docker/Resources/DockerResourceDetector.cs

```diff
@@ -28,6 +28,8 @@ namespace OpenTelemetry.Extensions.Docker.Resources;
 public class DockerResourceDetector : IResourceDetector
 {
     private const string FILEPATH = "/proc/self/cgroup";
+    private const string FILEPATHV2 = "/proc/self/mountinfo";
```

Comment:
I don't see any usage of `FILEPATHV2` in detector. I'd guess, that `ExtractContainerIdV2` should use it, instead of `FILEPATH`. Proper fix for it will also solve problem mentioned by @alexeypukhov , as that functions should check different file.

Reply:
Change has been made to check both versions in the Detect method and assuming the version being used in the BuildResource method. This way, the problem mentioned by @alexeypukhov would be solved as you mentioned since no container id would be able to be found when checking for cgroupv1 if cgroupv2 is being used.